examples/persistable.php

```diff
@@ -16,14 +16,12 @@
 
 class PersistableEntry implements Persistable
 {
-    /** @var ObjectId */
-    private $id;
+    private ObjectId $id;
```

Comment:
I went through the various type changes and confirmed they were accurate, but a second look certainly won't hurt.

Reply:
Good to me. Some missing property types from `ext-mongodb`, maybe due to outdated stubs.

When types are added to properties, the biggest risk is to have an error when trying to access an uninitialized property.

Reply:
Note: the missing property types weren't caught by Rector for some reason, and I only later realised that we had an exclusion in the phpcs config to prevent native types from being added. This has since been removed and types for properties are now enforced.